### DISCUSSION TITLE
Potentially what windows expresses an invalid socket as

### DISCUSSION THREAD

**Discussion by KianKhadempour** [djc/pyrtls#51] [General]
I noticed the `// TODO: no clue how Windows expresses an error here?` in `SessionState.__new__`. I tested what I think to be an invalid socket on Windows and the output was `-1`:
```py
>>> import socket
>>> sock = socket.socket.__new__(socket.socket)
>>> print(sock.detach())
-1
```
The problem is, `RawSocket` is of type `u64`, so you can't pattern match `-1`. Is my test wrong? If so, how can I test it?

**Comment by djc** [comment]
Thanks for looking at this!

Huh, interesting. Since `RawSocket` is just an alias for `u64` in the end, I guess we could `extract::<i64>()` instead, check for -1, then cast it to `u64` before passing to `from_raw_socket()`?
